pkg/model/lattice/iamauthpolicy.go

```diff
@@ -0,0 +1,60 @@
+package lattice
+
+import (
+	"fmt"
+
+	anv1alpha1 "github.com/aws/aws-application-networking-k8s/pkg/apis/applicationnetworking/v1alpha1"
+	"github.com/aws/aws-application-networking-k8s/pkg/utils"
+	"github.com/aws/aws-sdk-go/aws/arn"
+)
+
+type AuthPolicyKind string
+
+const (
+	GatewayAuthPolicy AuthPolicyKind = "gateway"
+	ServiceAuthPolicy AuthPolicyKind = "service"
+)
+
+type IAMAuthPolicy struct {
+	Kind         AuthPolicyKind
+	ResourceName string
+	Policy       string
+}
+
+type IAMAuthPolicyResource struct {
+	ARN  arn.ARN
+	Id   string
+	Name string
+}
+
+type IAMAuthPolicyStatus struct {
+	Resource IAMAuthPolicyResource
+	State    string
```

Comment:
What's this?

Reply:
IAMAuthPolicy - State. It's lattice state. I think our current alignment is to use xxxStatus models for lattice managers responses, so I would read it as it's state of IAMAuthPolicy in lattice